### PR TITLE
[special] Alias for special.expm1 and special.exp2

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: #50345

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -3177,12 +3177,14 @@ def reference_lgamma(x):
                    dtypesIfCUDA=floating_types_and(torch.float16),
                    assert_autodiffed=True),
     UnaryUfuncInfo('exp2',
+                   aliases=('special.exp2', ),
                    ref=np_unary_ufunc_integer_promotion_wrapper(np.exp2),
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:3181]: Have kept `np.exp2` as reference as it matches Pytorch promotion behavior for floats (similar for expm1).
`scipy.special.exp2` promotes float16 to float32 (thus there is a mismatch in result)

```python
>>> import torch
>>> import numpy as np
>>> import scipy.special
>>> import math
>>> t = torch.tensor([math.pi/2], dtype=torch.float16)
>>> torch.exp2(t)
tensor([2.9688], dtype=torch.float16)
>>> np.exp2(t.numpy())
array([2.969], dtype=float16)
>>> scipy.special.exp2(t.numpy())
array([2.9696903], dtype=float32)
```

**Comment by kshitij12345** [comment]
@mruberry Gentle Ping :)

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -3082,7 +3082,10 @@ def merge_dicts(*dicts):
 
 .. math::
     y_{i} = 2^{x_{i}}
+
+.. note:: Alias for :func:`torch.special.exp2`.
```
Comment (mruberry) [inline torch/_torch_docs.py:3086]: This is OK but eventually we should just replace these doc strings with the pointer to what they alias, like the arccos docs: https://pytorch.org/docs/master/generated/torch.arccos.html?highlight=alias.

File: torch/csrc/api/include/torch/special.h [hunk 3]
```diff
@@ -65,4 +65,28 @@ inline Tensor erfinv(const Tensor& self) {
...
 
+/// Computes the base two exponential function of :attr:`input`, elementwise
+/// See https://pytorch.org/docs/master/special.html#torch.special.exp2.
+///
+/// Example:
+/// ```
+/// auto t = torch::randn(128, dtype=kDouble);
+/// torch::special::exp2(t);
+/// ```
+inline Tensor exp2(const Tensor& self) {
```
Comment (mruberry) [inline torch/csrc/api/include/torch/special.h:76]: What about the out variants?

**Review by mruberry — APPROVED** [review]
Cool!

Some follow-up tasks:

- support out variants in C++ API
- aliasing should just have the docs of one function point to the other to avoid redundancy

**Pull request closed without merging**